macs/guest/apply.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+echo "apply started at $(date)" | nc -w0 -u 10.172.170.1 1514
+
+printf '\n*.*\t@10.172.170.1:1514\n' | tee -a /etc/syslog.conf
+pkill syslog
+pkill asl
+
+exec 3>&1
+exec 2> >(nc -u 10.172.170.1 1514)
+exec 1>&2
+
+PS4='${BASH_SOURCE}::${FUNCNAME[0]}::$LINENO '
+set -o pipefail
+set -ex
+date
+
+function finish {
+    set +e
+    cd /
+    sleep 1
+    umount -f /Volumes/CONFIG
+}
+trap finish EXIT
+
+cat <<EOF | tee -a /etc/ssh/sshd_config
+PermitRootLogin prohibit-password
+PasswordAuthentication no
+PermitEmptyPasswords no
+ChallengeResponseAuthentication no
+EOF
+
+launchctl stop com.openssh.sshd
+launchctl start com.openssh.sshd
+
+
+cd /Volumes/CONFIG
+
+cp -r ./etc/ssh/ssh_host_* /etc/ssh
+chown root:wheel /etc/ssh/ssh_host_*
+chmod 600 /etc/ssh/ssh_host_*
+cd /
+
+echo "%admin ALL = NOPASSWD: ALL" | tee /etc/sudoers.d/passwordless
+
+(
+    # Make this thing work as root
+    export USER=root
+    export HOME=~root
+    export ALLOW_PREEXISTING_INSTALLATION=1
+    env
+    curl https://nixos.org/releases/nix/nix-2.1.3/install > ~nixos/install-nix
+    chmod +rwx ~nixos/install-nix
+    cat /dev/null | sudo -i -H -u nixos -- sh ~nixos/install-nix --daemon
+)
+
+(
+    # Make this thing work as root
+    export USER=root
+    export HOME=~root
+
+    . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
+    env
+    ls -la /private || true
+    ls -la /private/var || true
+    ls -la /private/var/run || true
+    ln -s /private/var/run /run || true
+    nix-channel --add https://nixos.org/channels/nixos-19.03 nixpkgs
+    nix-channel --add https://github.com/LnL7/nix-darwin/archive/master.tar.gz darwin
+    nix-channel --update
+
+    sudo -i -H -u nixos -- nix-channel --add https://nixos.org/channels/nixos-19.03 nixpkgs
+    sudo -i -H -u nixos -- nix-channel --add https://github.com/LnL7/nix-darwin/archive/master.tar.gz darwin
+    sudo -i -H -u nixos -- nix-channel --update
```

Comment:
are all those channels being used? nixpkgs is quite big to download nowadays

Reply:
To be honest, I'm not sure if it is critical we have them for root and the user. This would be a good thing to figure out :)